engine/paymail_service_provider.go

```diff
@@ -182,14 +182,24 @@ func (p *PaymailDefaultServiceProvider) VerifyMerkleRoots(
 		}()
 	}
 
-	request := make([]chainstate.MerkleRootConfirmationRequestItem, 0, len(merkleRoots))
-	for _, m := range merkleRoots {
-		request = append(request, chainstate.MerkleRootConfirmationRequestItem{
-			MerkleRoot:  m.MerkleRoot,
-			BlockHeight: m.BlockHeight,
-		})
+	confirmations, err := p.client.Chain().VerifyMerkleRoots(ctx, merkleRoots)
```

Comment:
IMHO previous approach was better,
I mean, that we verify merkle roots inside the Chainstate (currently Chain), and not returning the response here.
From the "client" of the VerifyMerkleRoots method perspective, all I want to is to either have information that it's valid or it's not valid - and then probably a proper detailed error could be returned.

Reply:
Let's discuss it on huddle 😄 - because I have some points on that.

Reply:
I've adjusted it as we talked.
- Noe the VerifyMerkleRoots returns `(valid bool, err error)`